offer-manager-domain/src/test/groovy/life/qbic/portal/portlet/offers/OfferSpec.groovy

```diff
@@ -592,14 +598,14 @@ class OfferSpec extends Specification {
         List<ProductItem> items = [primaryAnalysis, projectManagement, sequencing, dataStorage, secondaryAnalysis]
         Offer offer = new Offer.Builder(customerWithAllAffiliations, projectManager, "Awesome Project", "An " +
                 "awesome project", items, affiliation).build()
-        def overheadSum = 0
+        BigDecimal overheadSum = 0
 
         when: "the net price is calculated"
```

Comment:
Could be done like this
```suggestion
        when: "the net price is calculated"
        def itemNet =  {
            BigDecimal.valueOf(it.totalPrice - it.quantityDiscount)
        }
        BigDecimal expectedValue = offer.getItems().collect(itemNet).sum() * overheadRatio
        overheadSum = offer.getOverheadSum()

        then: "the correct prices are taken into account"
        offer.selectedCustomerAffiliation.category == AffiliationCategory.EXTERNAL || offer.selectedCustomerAffiliation.category == AffiliationCategory.EXTERNAL_ACADEMIC
        hasRequiredPrecision(overheadSum, expectedValue)

        where: "the affiliation is"
```

Reply:
where did you define `hasRequiredPrecision ()`?

Reply:
```
    private static boolean hasRequiredPrecision(BigDecimal value, BigDecimal expectedValue) {
        return Math.abs(value - expectedValue) < MAX_NUMERIC_ERROR
    }
```
below the where block in a helper method. I could not add it to the suggestions because git interpreted changes as deletions and does not let you make suggestions on deletions.